lib/Server.js

```diff
@@ -118,7 +118,7 @@ class Server {
     this.setupDevMiddleware();
 
     // set express routes
-    routes(this.app, this.middleware, this.options);
```

Comment:
Why did you change this?

Reply:
@hiroppy To pass the server, so I can call server.invalidate()

Reply:
I'm -1 on this change because if you pass `this`, we can't understand what variables are used clearly.

Reply:
@hiroppy I tried `(this.app, this.middleware, this.options, this.invalidate)` and called `invalidate()` only but this resulted in `500 "Internal Server Error"` so what about changing `this` into `server` for better reading ex: 
```
const server = this;
routes(server);
```

Reply:
one thing that worked was moving 
```javascript
invalidate(callback) {
    if (this.middleware) {
      this.middleware.invalidate(callback);
    }
  }
``` 
to `routes.js` and make it only accessable via the api call, but that would require adding an option `invalidateCallBack` so we can pass it ex: 
```javascript
    this.invalidateCallBack = this.options.invalidateCallBack;
    // set express routes
    routes(this.app, this.middleware, this.options, this.invalidateCallBack);
```
and in `routes.js` 
```javascript
function routes(app, middleware, options, invalidateCallBack) {
  app.get('/invalidate', (_req, res) => {
    if(middleware)
      middleware.invalidate(invalidateCallBack)
    res.end();
  });
}
```